plugins/browser-plugin-web-vitals/src/utils.ts

```diff
@@ -0,0 +1,71 @@
+import { LOG } from '@snowplow/tracker-core';
+import { ReportCallback, WebVitals } from './types';
+
+/**
+ * Attach page listeners to collect the Web Vitals values
+ * @param {() => void} callback
+ */
+export function attachWebVitalsPageListeners(callback: () => void) {
+  // Safari does not fire "visibilitychange" on the tab close
+  // So we have 2 options: lose Safari data, or lose LCP/CLS that depends on "visibilitychange" logic.
+  // Current solution: if LCP/CLS supported, use `onHidden` otherwise, use `pagehide` to fire the callback in the end.
+  //
+  // More details: https://github.com/treosh/web-vitals-reporter/issues/3
+  const supportedEntryTypes = (PerformanceObserver && PerformanceObserver.supportedEntryTypes) || [];
+  const isLatestVisibilityChangeSupported = supportedEntryTypes.indexOf('layout-shift') !== -1;
+
+  if (isLatestVisibilityChangeSupported) {
+    const onVisibilityChange = () => {
+      if (document.visibilityState === 'hidden') {
+        callback();
+        removeEventListener('visibilitychange', onVisibilityChange, true);
```

Comment:
Nit: can we use `window.addEventListener` / `window.removeEventListener` or `addEventListener` / `removeEventListener` consistently?

Reply:
Done